version: '3.8'

networks:
  django-app-net:

volumes:
  postgres_data:

services:
  db:
    image: postgres:15
    container_name: postgres_db
    environment:
      - POSTGRES_DB=${DATABASE_NAME}
      - POSTGRES_USER=${DATABASE_USER}
      - POSTGRES_PASSWORD=${DATABASE_PASSWORD}
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - django-app-net
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DATABASE_USER}"]
      interval: 30s
      timeout: 10s
      retries: 5

  web:
    build:
      context: .
    container_name: django_clinic
    command: >
      sh -c "
      ./wait-for-it.sh postgres_db:5432 -- 
      python manage.py migrate &&
      python createsuperuser.py &&
      python manage.py makemigrations &&
      python manage.py migrate &&
      gunicorn clinic.wsgi:application --bind 0.0.0.0:8000 --timeout 120
      "
    volumes:
      - .:/app
    ports:
      - "8000:8000"
    environment:
      - DATABASE_URL=postgres://postgres:postgres@postgres_db:5432/django_clinic_db
      - DJANGO_SUPERUSER_USERNAME=${DJANGO_SUPERUSER_USERNAME}
      - DJANGO_SUPERUSER_EMAIL=${DJANGO_SUPERUSER_EMAIL}
      - DJANGO_SUPERUSER_PASSWORD=${DJANGO_SUPERUSER_PASSWORD}
    depends_on:
      db:
        condition: service_healthy
    networks:
      - django-app-net
    env_file:
      - .env
